transformer_lens/components.py

```diff
@@ -593,6 +600,27 @@ def forward(
             )
             / self.attn_scale
         )  # [batch, head_index, query_pos, key_pos]
+
+        # alibi encoding before applying causal mask
+        if self.cfg.positional_embedding_type == 'alibi':
+            #TODO: not sure about the side effect of not using standard, double check
+            batch_size = attn_scores.size(0)
+            seq_len = attn_scores.size(-2) 
```

Comment:
Should this be -1? Note that when generating text the attention scores are *not* square

Reply:
Yes it should be set to key_length, changed to -1. Thanks!